### ISSUE TITLE
Support custom non-standard DOM events

### ISSUE THREAD

**Issue by Gozala** [facebook/react#4645]
Please add support for custom non-standard DOM events, ideally through a public API but if that's a "no go"  at least some way that would not require forking the code base.

Here is my attempt of exposing non-standard events after getting some insights on IRC, unfortunately it still does not work (as event listeners are not registered) and I am not sure if there is more whitelists somewhere that skips those custom events:
https://gist.github.com/Gozala/a977973957fcee9c5671

P.S.: Any help in changing this snippet to make it work is also greatly appreciated

**Comment by daukantas** [comment]
I hereby add my support  to custom non-standard DOM events.

**Comment by louisremi** [comment]
I recommend taking inspiration from the battle-tested [*special event* API](http://benalman.com/news/2010/03/jquery-special-events/) of jQuery. It has allowed devs to both normalize events across browsers ([wheel events](https://github.com/louisremi/jquery.wheel.js/blob/master/src/jquery.wheel.js) for example), and support new kind of events such as triple click (not saying this one's a good idea).

**Comment by Gozala** [comment]
Is there any chance to get any kind of feedback on this ?

**Comment by rpominov** [comment]
This may be related #4751

**Comment by sebmarkbage** [comment]
This has been asked for many times before. See discussion in #4751.

It might make sense to allow any function to pass through and become a "simple event plugin" by default. That way there is no whitelist necessarily. This would make it a breaking change if we ever added custom logic to that particular event. Which we likely will. However, that's the stance we've already taking with other attributes in general.

**Comment by Gozala** [comment]
> It might make sense to allow any function to pass through and become a "simple event plugin" by default. That way there is no whitelist necessarily. This would make it a breaking change if we ever added custom logic to that particular event. Which we likely will. However, that's the stance we've already taking with other attributes in general.

You could require a specific "simple event plugin" to avoid breaking changes, which could be as simple as:

```js
html.div({ onClick: handleSyntheticClick, onNativeClick: handleNativeClick })
```

Or more complex but robust as:

```js
html.div({
  onClick: handleSyntheticClick,
  [React.on('click')]: handleNativeClick
})
```

**Comment by nhunzaker** [comment]
@sebmarkbage Sort of spurred by https://github.com/facebook/react/issues/7901, but I've been wanting custom events for a while and had some prior work stripping out the attribute whitelist (https://github.com/facebook/react/pull/7311).

So I took a stab:
https://github.com/nhunzaker/react/pull/2

I've PR'ed this to myself (against #7311). I'm hesitant to submit it, but I think I'm happy with it. There was a lot of event name munging already happening, I was able to lower that, move most of it to a central location, and lean on the existing `SimpleEventPlugin` as you'd suggested.

Does this feel like the right track?
